services/app-api/utils/validation/validation.ts

```diff
@@ -2,6 +2,22 @@ import * as yup from "yup";
 import { nested, endDate, schemaMap } from "./schemaMap";
 import { AnyObject } from "../types/types";
 import { error } from "../constants/constants";
+import { APIGatewayEvent } from "aws-lambda";
+
+// validate report keys
+export const validReport = (
+  event: APIGatewayEvent,
+  apiMethod: string | undefined
+) => {
+  const pathParameters = event.pathParameters!;
+  if (apiMethod === "create") {
```

Comment:
Thought: This totally works. We could also do something by making it agnostic to what kind of handler it is and what parameters are required by passing them in (think an array of strings that we pass in, called `requiredParams`).  That array, say `["id", "state", "reportType"]` or `["id", "state"]`, could then be checked to make sure it exists. Something like (with silly types so it highlights the syntax lol):

```js
export const hasRequiredPathParams = ( eventParams: AnyObject, requiredParams: ArrayOfStrings ) => 
requiredParams.every((requiredParam: SingleString) => eventParams[requiredParam]);

// then calling it would look like this
const requiredParams = ["id", "state", "reportType"];
if (!hasRequiredPathParams(event.pathParameters, requiredParams) {
  // fail hard
} 
// continue!
```

Reply:
Good call, I moved this to its own utility under `dynamo` and that makes way more sense.